### PR TITLE
Add donor_end to player object, use pytimeparse consistently

### PR DESCRIPTION
**Pull request by 7ez**
In the current state, the command won't run as there is no `donor_end` in the Player object.

### CODE REVIEW

**Comment by 7ez** [comment]
Code was also optimised a bit if you can even call it that

File: app/commands.py [hunk 1]
```diff
@@ -1410,6 +1373,29 @@ async def rmpriv(ctx: Context) -> Optional[str]:
     await t.remove_privs(bits)
     return f"Updated {t}'s privileges."
 
+@command(Privileges.DEVELOPER, hidden=True)
```
Comment (minisbett) [inline app/commands.py:1379]: I think admins should be able to assign donator privileges, I don't see a benefit in restricting the access to developers.

**Review by minisbett — CHANGES_REQUESTED** [review]

File: app/commands.py [hunk 2]
```diff
@@ -905,6 +900,7 @@ async def user(ctx: Context) -> Optional[str]:
         (
             f'[{"Bot" if p.bot_client else "Player"}] {p.full_name} ({p.id})',
             f"Privileges: {priv_list}",
+            f"Donator ends: {timeago.format(p.donor_end)}",
```
[the remark this answers is not in the document]
Comment (7ez) [reply app/commands.py:903]: alr

File: app/commands.py [hunk 3]
```diff
@@ -1411,6 +1376,31 @@ async def rmpriv(ctx: Context) -> Optional[str]:
...
+        timespan += int(time.time())
+    else:
+        timespan += t.donor_end
+
+    await app.state.services.database.execute(
+        "UPDATE users SET donor_end = :end WHERE id = :user_id",
+        {"end": timespan, "user_id": t.id},
+    )
+
+    return f"Added {''.join(ctx.args[1:])} of donator status to {t}."
```
Comment (cmyui) [inline app/commands.py:1401]: should be just ctx.args[1]

File: app/commands.py [hunk 4]
```diff
@@ -1411,6 +1376,31 @@ async def rmpriv(ctx: Context) -> Optional[str]:
...
+@command(Privileges.DEVELOPER, hidden=True)
+async def givedonator(ctx: Context) -> Optional[str]:
+    """Gives donator to a specified player (by name) for a specified amount of time, such as '3h5m'."""
+    if len(ctx.args) < 2:
+        return "Invalid syntax: !givedonator <name> <duration>"
+
+    if not (t := await app.state.sessions.players.from_cache_or_sql(name=ctx.args[0])):
+        return "Could not find user."
+
+    if not (timespan := timeparse("".join(ctx.args[1:]))):
```
Comment (cmyui) [inline app/commands.py:1388]: should be just ctx.args[1] - `len(ctx.args) < 2` is asserted above

File: app/commands.py (hunk 3 above)
Comment (7ez) [reply app/commands.py:1401]: i mean people may use spaces so incase they do it'll still support it

File: app/commands.py (hunk 4 above)
Comment (7ez) [reply app/commands.py:1388]: that only checks if theres less than 2 args lol

Comment (cmyui) [reply app/commands.py:1388]: Oh right, didn't realize pytimeparse supported that
```py
>>> pytimeparse.parse(' 3h 5m')
11100
```

in this case, just make sure we're being consistent across all commands doing time parsing with this - if we allow spaces in one, we should allow spaces in all

Comment (7ez) [reply app/commands.py:1388]: i'd have to move the delay at the end on the shutdown command then

Comment (cmyui) [reply app/commands.py:1388]: ah sorry shutdown is okay as-is then, just worry about user-facing commands, dev commands are lower prio

Comment (7ez) [reply app/commands.py:1388]: what about silence though

Comment (7ez) [reply app/commands.py:1388]: do i just move duration to the end?

Comment (7ez) [reply app/commands.py:1388]: ill just leave it for now

File: app/commands.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (minisbett) [reply app/commands.py]: I wanted to do that once this pr is merged

[the remark this answers is not in the document]
Comment (7ez) [reply app/commands.py]: its already been done though

**Comment by 7ez** [comment]
@cmyui all good now 👍

**Review by cmyui — APPROVED** [review]

**Pull request merged**